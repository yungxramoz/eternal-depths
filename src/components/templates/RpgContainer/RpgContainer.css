.full-page {
  width: 100vw;
  height: 100vh;
}
