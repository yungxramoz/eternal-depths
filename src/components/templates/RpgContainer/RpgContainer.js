import React from 'react'
import './RpgContainer.css'

const RpgContainer = ({
  framed = true,
  golden,
  golden2,
  grey,
  fullPage,
  children,
  className,
  bgImg,
}) => {
  const containerClasses = ['rpgui-container']
  if (framed) containerClasses.push('framed')
  if (golden) containerClasses.push('framed-golden')
  if (golden2) containerClasses.push('framed-golden-2')
  if (grey) containerClasses.push('framed-grey')
  if (fullPage) containerClasses.push('full-page')
  if (className) containerClasses.push(className)

  let bgImgStyle = {}
  if (bgImg) {
    bgImgStyle = {
      backgroundImage: `url(${bgImg})`,
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
      backgroundPosition: 'bottom',
    }
  }

  return (
    <div className={containerClasses.join(' ')} style={bgImgStyle}>
      {children}
    </div>
  )
}

export default RpgContainer
