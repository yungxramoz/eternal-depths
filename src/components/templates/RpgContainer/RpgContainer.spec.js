import React from 'react'
import TestRenderer from 'react-test-renderer'
import RpgContainer from './RpgContainer'

describe('RpgContainer', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(<RpgContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('applies correct classes based on props', async () => {
    const testRenderer = TestRenderer.create(
      <RpgContainer
        framed
        golden
        golden2
        grey
        fullPage
        className="custom-class"
      />,
    )
    const testInstance = testRenderer.root
    const div = await testInstance.findByType('div')
    expect(div.props.className).toEqual(
      'rpgui-container framed framed-golden framed-golden-2 framed-grey full-page custom-class',
    )
  })
})
