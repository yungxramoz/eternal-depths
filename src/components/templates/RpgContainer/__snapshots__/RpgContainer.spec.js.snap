// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RpgContainer applies background image based on bgImg prop: RpgContainer Background Image 1`] = `
<body>
  <div>
    <div
      class="rpgui-container container framed"
      style="background-image: url(test.png); background-repeat: no-repeat; background-size: cover; background-position: bottom;"
    />
  </div>
</body>
`;

exports[`RpgContainer renders correctly: RpgContainer 1`] = `
<body>
  <div>
    <div
      class="rpgui-container container framed"
    />
  </div>
</body>
`;
